dot/sync/disjoint_block_set_test.go

```diff
@@ -195,3 +204,24 @@ func TestDisjointBlockSet_getReadyDescendants_blockNotComplete(t *testing.T) {
 	require.Equal(t, block1.ToBlockData(), ready[0])
 	require.Equal(t, block2.ToBlockData(), ready[1])
 }
+
+func TestDisjointBlockSet_ClearBlocks(t *testing.T) {
+	s := newDisjointBlockSet(pendingBlocksLimit)
+
+	testHashA := common.Hash{0}
+	testHashB := common.Hash{1}
+
+	s.blocks[testHashA] = &pendingBlock{
+		hash:    testHashA,
+		clearAt: time.Unix(1000, 0),
+	}
+	s.blocks[testHashB] = &pendingBlock{
+		hash:    testHashB,
+		clearAt: time.Now().Add(ttl * 2),
+	}
+
+	s.clearBlocks()
+	require.Equal(t, 1, len(s.blocks))
+	_, has := s.blocks[testHashB]
```

Comment:
Should the returned value be asserted as well :thinking:

Reply:
we just need to check that the block with hash b exists and a doesn't, `clearBlocks` doesn't modify blocks

Reply:
> doesn't modify blocks

Yes, we should check that :smile: Just in case :tm: (also in case someone modifies the code in the future)
The deeper the test the better!